firebase-perf/src/test/java/com/google/firebase/perf/transport/RateLimiterTest.java

```diff
@@ -84,41 +91,215 @@ public void testRateLimitImpl() {
     // allow 2 logs every minute. token bucket capacity is 2.
     // clock is 0, token count is 2.
     RateLimiterImpl limiter =
-        new RateLimiterImpl(2.0d / 60, 2, mClock, mockConfigResolver, NETWORK, false);
+        new RateLimiterImpl(TWO_TOKENS_PER_MINUTE, 2, mClock, mockConfigResolver, NETWORK, false);
```

Comment:
What I'm seeing here is that with 2 tokens per minute, 4 requests in the first minute all succeed, which doesn't seem to make sense to me.

Reply:
The way I am seeing it is: if we have a full bucket of 2 tokens at time 0, and we replenish 2 tokens per minute, we will have four tokens available to be used in the first minute. Otherwise, are we saying we shouldn't replenish the bucket until it's empty?

Reply:
I think this is slightly confusing since then we have double the configured capacity on the first minute. I think the idea to have an initial capacity of 2 is so that we don't start by throttling requests, not necessarily so we can have double the capacity during the first minute.

Reply:
Perhaps we can look at it differently. Even though conceptually to us, the rate is 2 tokens per minute, the implementation actually views the rate as token per second and calculates how many to refill per second. So in this example, the rate per second is 1/30. If viewed this way, we have multiplied the capacity during the first second by 60 times by allowing 2 events to go through within the first second but the rate per second is 1/30.

Reading more about the token bucket algorithm (wiki: https://en.wikipedia.org/wiki/Token_bucket), the capacity is there to allow a burst of events while the rate is there to regulate the average rate of events in the long run. I think that more events will be allowed to pass through when the capacity is more than the rate; capacity is the cap for the max number of events that can pass through, not the rate. For example, if the capacity is 500, and our rate is 2 per minute, we would allow all 500 events to go through in the first minute.